src/dbnode/persist/fs/write.go

```diff
@@ -161,6 +161,11 @@ func (w *writer) Open(opts DataWriterOpenOptions) error {
 	w.currIdx = 0
 	w.currOffset = 0
 	w.err = nil
+	// This happens after writing the previous set of files index files, however, do it
```

Comment:
mind grouping the reseting code into a new `func (w *writer) reset()` fn

Reply:
tangentially - should `indexEntries` be bounded/randomly re-allocated if it exceeds certain size to reduce memory usage? not for this PR, just curious myself.

Reply:
@prateek sure

Reply:
@vdarulis Theoretically yeah it probably should. In practice we only keep one of these around per node (for the most part) and the size of this slice will never exceed the number of series for a given block/shard combination so in practice its usually in the 10s of thousands tops and each item in the slice is not *that* big so I think its really unlikely it would become an issue. There are definitely other things like this in the code-base though where we're more paranoid because they could become an issue if left unchecked